### PR TITLE
added test on protected_packages

### PR DESCRIPTION
**Pull request by ignatenkobrain**

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
@jsilhan 
```
Feature: DNF/Behave test (protected_packages test) # behave/protected_packages.feature:1

  Scenario: Removal of installed protected package                                                                                     # behave/protected_packages.feature:3
    Given I use the repository "test-1"                                                                                                # behave/steps/test_behave.py:63
    When I execute "dnf" command "install -y TestA" with "success"                                                                     # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State     | Packages     |
      | installed | TestA, TestB |
    When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "failure"                                    # behave/steps/test_behave.py:82
      Assertion Failed: The option failure is not allowed option for expected result of command. Allowed options are "success" and "fail"
      Captured stdout:
      Error: The operation would result in removing the following protected packages: TestA

    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: TestA" # None
    When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "failure"                                    # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: TestA" # None
    When I execute "dnf" command "-y remove TestB --setopt=protected_packages=TestA" with "failure"                                    # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |
    When I execute "dnf" command "-y remove dnf" with "failure"                                                                        # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages |
      | absent | dnf      |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: dnf"   # None
    When I create a file "/etc/yum/protected.d/donotremove" with content: "TestA"                                                      # None
    When I execute "dnf" command "-y remove TestA with "failure"                                                                       # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |


You can implement step definitions for undefined steps with these snippets:

@when(u'I execute "dnf" command "-y remove TestA with "failure"')
def step_impl(context):
    raise NotImplementedError(u'STEP: When I execute "dnf" command "-y remove TestA with "failure"')


Failing scenarios:
  behave/protected_packages.feature:3  Removal of installed protected package

0 features passed, 1 failed, 0 skipped
0 scenarios passed, 1 failed, 0 skipped
3 steps passed, 1 failed, 12 skipped, 1 undefined
Took 0m3.771s
```

**Comment by ignatenkobrain** [comment]
even after quick fix it still doesn't work:
```
Feature: DNF/Behave test (protected_packages test) # behave/protected_packages.feature:1

  Scenario: Removal of installed protected package                                                                                     # behave/protected_packages.feature:3
    Given I use the repository "test-1"                                                                                                # behave/steps/test_behave.py:63
    When I execute "dnf" command "install -y TestA" with "success"                                                                     # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State     | Packages     |
      | installed | TestA, TestB |
    When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "fail"                                       # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State  | Packages     |
      | absent | TestA, TestB |
      Assertion Failed: The number of found packages 1 (packages - TestA-1.0.0-1.noarch) for string "TestA" differs from expected 0
      Captured stdout:
      Error: The operation would result in removing the following protected packages: TestA

    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: TestA" # None
    When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "fail"                                       # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: TestA" # None
    When I execute "dnf" command "-y remove TestB --setopt=protected_packages=TestA" with "fail"                                       # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |
    When I execute "dnf" command "-y remove dnf" with "fail"                                                                           # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages |
      | absent | dnf      |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: dnf"   # None
    When I create a file "/etc/yum/protected.d/donotremove" with content: "TestA"                                                      # None
    When I execute "dnf" command "-y remove TestA with "fail"                                                                          # None
    Then transaction changes are as follows                                                                                            # None
      | State  | Packages     |
      | absent | TestA, TestB |


Failing scenarios:
  behave/protected_packages.feature:3  Removal of installed protected package

0 features passed, 1 failed, 0 skipped
0 scenarios passed, 1 failed, 0 skipped
4 steps passed, 1 failed, 11 skipped, 1 undefined
Took 0m3.800s

You can implement step definitions for undefined steps with these snippets:

@when(u'I execute "dnf" command "-y remove TestA with "fail"')
def step_impl(context):
    raise NotImplementedError(u'STEP: When I execute "dnf" command "-y remove TestA with "fail"')

```

File: dnf-docker-test/features/protected_packages.feature [hunk 1]
```diff
@@ -7,31 +7,31 @@ Scenario: Removal of installed protected package
    | State        | Packages      |
    | installed    | TestA, TestB  |
 
- When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "failure"
+ When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "fail"
  Then transaction changes are as follows
    | State        | Packages     |
    | absent       | TestA, TestB |
```
Comment (j-mracek) [inline dnf-docker-test/features/protected_packages.feature]: | present | TestA, TestB |

The packages were installed in step above, and not changed be previous step, therefore they are still present. Absent means that they are not installed, but they wer not installed even before last dnf command.

**Comment by ignatenkobrain** [comment]
```
Feature: DNF/Behave test (protected_packages test) # behave/protected_packages.feature:1

  Scenario: Removal of installed protected package                                                                                     # behave/protected_packages.feature:3
    Given I use the repository "test-1"                                                                                                # behave/steps/test_behave.py:63
    When I execute "dnf" command "install -y TestA" with "success"                                                                     # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State     | Packages     |
      | installed | TestA, TestB |
    When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "fail"                                       # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State   | Packages     |
      | present | TestA, TestB |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: TestA" # behave/steps/test_behave.py:184
    When I execute "dnf" command "-y remove TestA --setopt=protected_packages=TestA" with "fail"                                       # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State   | Packages     |
      | present | TestA, TestB |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: TestA" # behave/steps/test_behave.py:184
    When I execute "dnf" command "-y remove TestB --setopt=protected_packages=TestA" with "fail"                                       # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State   | Packages     |
      | present | TestA, TestB |
    When I execute "dnf" command "-y remove dnf" with "fail"                                                                           # behave/steps/test_behave.py:82
    Then transaction changes are as follows                                                                                            # behave/steps/test_behave.py:136
      | State   | Packages |
      | present | dnf      |
    And line from "stderr" should "start" with "Error: The operation would result in removing the following protected packages: dnf"   # behave/steps/test_behave.py:184
    When I create a file "/etc/yum/protected.d/donotremove" with content: "TestA"                                                      # behave/steps/test_behave.py:113
    When I execute "dnf" command "-y remove TestA" with "fail"                                                                         # behave/steps/test_behave.py:82
      Assertion Failed: Return code was 0, but expected non zero (fail)
      Captured stdout:
      Error: The operation would result in removing the following protected packages: TestA
      
      Error: The operation would result in removing the following protected packages: TestA
      
      Error: package TestA-1.0.0-1.noarch requires TestB, but none of the providers can be installed
      
      Error: The operation would result in removing the following protected packages: dnf

    Then transaction changes are as follows                                                                                            # None
      | State   | Packages     |
      | present | TestA, TestB |
    When I execute "dnf" command "-y remove TestB" with "fail"                                                                         # None
    Then transaction changes are as follows                                                                                            # None
      | State   | Packages     |
      | removed | TestA, TestB |


Failing scenarios:
  behave/protected_packages.feature:3  Removal of installed protected package

0 features passed, 1 failed, 0 skipped
0 scenarios passed, 1 failed, 0 skipped
15 steps passed, 1 failed, 3 skipped, 0 undefined
Took 0m9.681s
```

**Comment by jsilhan** [comment]
Igor, test this please. It should work now.

**Comment by ignatenkobrain** [comment]
yes, it even works.

**Pull request merged**